app/controllers/AdditionalMetadataNavigationController.scala

```diff
@@ -32,20 +33,34 @@ class AdditionalMetadataNavigationController @Inject() (
       .toList
       .filter(_ != "csrfToken")
       .map(UUID.fromString)
-    if (metadataType == "closure") {
-      val fileFilters = FileFilters(None, Option(fileIds), None)
-      consignmentService
-        .getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Option(fileFilters))
-        .map(consignment => {
-          val areAllClosed = consignmentService.areAllFilesClosed(consignment)
-          if (areAllClosed) {
-            Redirect(routes.AddClosureMetadataController.addClosureMetadata(consignmentId, fileIds))
-          } else {
-            Redirect(routes.AdditionalMetadataClosureStatusController.getClosureStatusPage(consignmentId, fileIds))
-          }
-        })
+
+    if (fileIds.nonEmpty) {
+      if (metadataType == "closure") {
+        val fileFilters = FileFilters(None, Option(fileIds), None)
+        consignmentService
+          .getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Option(fileFilters))
+          .map(consignment => {
+            val areAllClosed = consignmentService.areAllFilesClosed(consignment)
+            if (areAllClosed) {
+              Redirect(routes.AddClosureMetadataController.addClosureMetadata(consignmentId, fileIds))
+            } else {
+              Redirect(routes.AdditionalMetadataClosureStatusController.getClosureStatusPage(consignmentId, fileIds))
+            }
+          })
+      } else {
+        Future(Redirect(routes.AdditionalMetadataSummaryController.getSelectedSummaryPage(consignmentId, fileIds)))
+      }
     } else {
-      Future(Redirect(routes.AdditionalMetadataSummaryController.getSelectedSummaryPage(consignmentId, fileIds)))
+      getCachedFiles(consignmentId, metadataType, request).map(allFiles => {
+        BadRequest(views.html.standard.additionalMetadataNavigation(consignmentId, request.token.name, allFiles, metadataType, displayError = true))
+      })
     }
   }
+
+  private def getCachedFiles(consignmentId: UUID, metadataType: String, request: Request[AnyContent]): Future[ConsignmentService.File] = {
```

Comment:
Does this cache need distinguishing between the different metadata types? Its always going to be the same regardless of the metadata type, as it contains all the filed ids for the consignment.

Reply:
Yeah, I was thinking if we are gonna show the green tick then this will be useful to distinguish between the different metadata types. But, if we decide to use not to show or use a different approach then we will remove the metadata type from the cache key.